src/org.hdfgroup.object/hdf/object/h5/H5CompoundAttr.java

```diff
@@ -1855,67 +1871,63 @@ public Object AttributeCommonIO(long attr_id, H5File.IO_TYPE ioType, Object objB
      * @throws Exception
      *             if the data can not be retrieved
      */
+    @Override
     public Object AttributeSelection() throws Exception {
         return originalBuf;
-//        H5Datatype dsDatatype = (H5Datatype) getDatatype();
-//        Object theData = H5Datatype.allocateArray(dsDatatype, (int)nPoints);
-//        if (dsDatatype.isText() && convertByteToString && (theData instanceof byte[])) {
-//        log.trace("AttributeSelection(): isText: converting byte array to string array");
-//        theData = byteToString((byte[]) theData, (int) dsDatatype.getDatatypeSize());
-//    }
-//    else if (dsDatatype.isFloat() && dsDatatype.getDatatypeSize() == 16) {
-//        log.trace("AttributeSelection(): isFloat: converting byte array to BigDecimal array");
-//        theData = dsDatatype.byteToBigDecimal(0, (int)nPoints, (byte[]) theData);
-//    }
-//    else if (dsDatatype.isArray() && dsDatatype.getDatatypeBase().isFloat() && dsDatatype.getDatatypeBase().getDatatypeSize() == 16) {
-//        log.trace("AttributeSelection(): isArray and isFloat: converting byte array to BigDecimal array");
-//        long[] arrayDims = dsDatatype.getArrayDims();
-//        int asize = (int)nPoints;
-//        for (int j = 0; j < arrayDims.length; j++) {
-//            asize *= arrayDims[j];
-//        }
-//        theData = ((H5Datatype)dsDatatype.getDatatypeBase()).byteToBigDecimal(0, asize, (byte[]) theData);
-//    }
-//    Object theOrig = originalBuf;
+        //        H5Datatype dsDatatype = (H5Datatype) getDatatype();
```

Comment:
Looks like below is just a bunch of leftover stuff?

Reply:
Or it might be unimplemented for compounds - I had some things working for scalar but not for compounds.